Automattic/Automattic-Tracks-iOS
# Update Sentry to v6

Updates the Sentry library to v6, which enables:
- Session tracking correlated with crashes to give us better crash rate data
- Support for building the project on arm64 macs

Unfortunately, the v6 library takes away our ability to manually send events to Sentry, so this PR adds that ability back by using the API directly.

Previously to this PR, testing Sentry integration was always a bit tricky because we'd need to integrate into a host app and make weird changes to it in order to test the integration.

This PR adds the ability to test Sentry functionality from the example project.

**To Test:**
- `cd` into the `TracksDemo` directory and run `bundle exec pod install`
- Create the `Shared/Secrets.swift` file:
   - For automatticians: copy the file from our mobile secrets repo (under `iOS/tracks-demo`). You might need to update (`git pull`) the mobile secrets repo first.
   - For non-automatticians: copy the file `Shared/Secrets.example.swift` and fill in the missing details
   - Feel free to customize your email address :)
   - _Note: In a future PR, we'll integrate the `configure` functionality to automate this, but it's not part of this PR in order to avoid making it any larger._
- Run the sample project on iOS. Go to the "Crashes" tab and try "Send Test Event" and "Send Error and Wait". Ensure that both events show up in the Sentry `tracks-demo` project with an associated stack trace.
- Run the same test on the macOS project. 

**Additional M1-specific tests:**
- Try building for `Any iOS Device`
- Try building for `My Mac`, and `Any Mac (Apple Silicon, Intel)` [ note: you may need to clean your build directory first – this tends to fail otherwise – I assume this is an Xcode bug ]

_I'd love one review on this from a mobile infrastructure engineer as well as a review from someone who has M1 hardware to test with._
